### DISCUSSION TITLE
Revisions in SQL ETL

### DISCUSSION THREAD

**Discussion by brunk01** [ravendb/ravendb#12219] [Q&A]
Googled it, didn't seem to find it. In RavenDB 5.1, is there a way to access the revisions in a SQL ETL task? I'd like to project the number of revisions to a SQL table. Thinking it might be accessible via the metadata similar to attachments, I tried this without success:

var revisions = this['@metadata']['@revisions'].Count;
loadToDocumentMetadataProjection(
{
ID: this.LegacyDocumentId,
RevisionCount: revisions
}

**Comment by arekpalinski** [comment]
Hi,

There is no option to get the count of revisions of a given document in ETL script at this point. That's something we might consider to add via a dedicated method accessible in the script. Can you tell us more about your use case?

**Reply by brunk01** [reply]
We are replacing an existing SQLServer based custom doc management system with Raven using Raven's revisions to manage change tracking. We used the number of revisions in the existing system to project the data for reporting and simply used the number of revisions to have a sense of "version" for reporting. We'd like to replicate this with Raven's SQL ETL

**Reply by arekpalinski** [reply]
Don't know if that can help you but since version 5,1 we have the following method available in API:

`await session.Advanced.Revisions.GetCountForAsync(company.Id)`

So maybe you could store it externally in a related document and then load it inside ETL script.

**Reply by brunk01** [reply]
Feature request? Something like this['@metadata']['@revisions'] as is done with attachments?

**Reply by arekpalinski** [reply]
The revisions aren't part of metadata, so more likely we'll expose it via a dedicated method like `getRevisionsCount()`. I have opened the ticket for that: https://issues.hibernatingrhinos.com/issue/RavenDB-16751
